### PR TITLE
[Merged by Bors] - Update linters.md with info about `cargo ci` xtask

### PR DESCRIPTION
**Pull request by CleanCut**
Update `linters.md` with info about `cargo ci` xtask as per https://github.com/bevyengine/bevy/pull/1463#pullrequestreview-593747890

### CODE REVIEW

**Comment by CleanCut** [comment]
@alice-i-cecile You're quick! By the time I thought to go add the labels, you'd already done it. 😄

**Review by mockersf — APPROVED** [review]

**Comment by DJMcNab** [comment]
I don't actually think that command works

That is, we don't actually have a .cargo/config.toml to add the command.

**Comment by MinerSebas** [comment]
I would rewrite the File like this:

---

## [xtask](https://github.com/matklad/cargo-xtask)

For simple cross-platform rust linting run the following command:

```bash
cargo run --package ci
```

### [rustfmt](https://github.com/rust-lang/rustfmt)

Is automatically run through [xtask](#xtask) (which also runs other checks) or manually with this command:

```bash
cargo fmt --all
```

### [Clippy](https://github.com/rust-lang/rust-clippy)

Can be automatically be run with the [xtask](#xtask) (which also runs other checks) or manually with this command:

```bash
cargo clippy --workspace --all-targets --all-features -- -D warnings -A clippy::type_complexity -A clippy::manual-strip
```

---

This PR should also wait until #1508 is merged or closed, as it switches away from xtask.

**Comment by CleanCut** [comment]
> I don't actually think that command works
> 
> That is, we don't actually have a .cargo/config.toml to add the command.

Thank you for finding that!  No wonder the command alias wasn't working in CI!  `.cargo/config.toml` is [listed in .gitignore](https://github.com/bevyengine/bevy/blob/bc4fe9b186db3f32eef38fc4241289df480fa949/.gitignore#L6)!!!  I'm surprised I didn't see anyone else reporting that it was broken for them.  I did add `.cargo/config.toml` locally and it's been working fine _for me_, but it never made it into my PRs. I hadn't seen that my "git add -A" didn't actually add it.

I added the file to #1508

**Comment by CleanCut** [comment]
> This PR should also wait until #1508 is merged or closed, as it switches away from xtask.

@MinerSebas I don't think this documentation update needs to be blocked on #1508. "`xtask`" is a pattern of using a cargo command alias and a Rust project to do something that would have been traditionally done in `bash`. #1508 preserves that pattern of using a Rust project and cargo alias (in fact, #1508 fixes the broken alias part), it just uses a different library for executing external commands (`duct` instead of `xshell`) to make it easier to cut down on unwanted warnings.

My takeaway from your content suggestion is that the "xtask" terminology is confusing. That's probably true -- there's no need for the doc to specify the implementation pattern we're using to accomplish things.  I've pushed up a new commit to try to simplify the wording further. What do you think of the updated wording?

**Comment by DJMcNab** [comment]
Personally, I'd prefer if we used the literal term `xtask` for our tasks

Then we could have `cargo xtask fmt` `cargo xtask ci`, `cargo xtask clippy` for example.
